Doc/howto/remote_debugging.rst

```diff
@@ -0,0 +1,335 @@
+.. _remote-debugging:
+
+Remote Debugging Attachment Protocol
+====================================
+
+This section explains the low-level protocol that allows external code to inject and execute
+a Python script inside a running CPython process.
+
+This is the mechanism implemented by the :func:`sys.remote_exec` function, which
+instructs a remote Python process to execute a ``.py`` file. This section is not about using that
+function, instead, it explains how the underlying protocol works so that it can be
+reimplemented in any language.
+
+The protocol assumes you already know the process you want to target and the code you want it to run.
```

Comment:
```suggestion
The protocol takes as input data the PID of the target process and a file containing the code to be executed.
```

In general, I recommend to not talk to a person (you) but in generic terms. We are writing a spec more than anything so it needs to read a bit more technical :)

Reply:
Ah instead I think we should drop this line and merge it with the following ones